core/ssh_keys_test.go

```diff
@@ -0,0 +1,453 @@
+			obj:  &sshKeysResponseBody{},
+		},
+		{
+			name: "ssh_key",
+			obj: &sshKeysResponseBody{
+				SSHKey: &AuthSSHKey{},
+			},
+		},
+		{
+			name: "ssh_keys",
+			obj: &sshKeysResponseBody{
+				SSHKeys: []*AuthSSHKey{{}},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func Test_AuthSSHKeyProperties_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *AuthSSHKey
+	}{
+		{
+			name: "empty",
+			obj:  &AuthSSHKey{},
+		},
+		{
+			name: "full",
+			obj: &AuthSSHKey{
+				ID:          "a",
+				Name:        "b",
+				Fingerprint: "c",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func Test_SSHKeyRef_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *SSHKeyRef
+	}{
+		{
+			name: "empty",
+			obj:  &SSHKeyRef{},
+		},
+		{
+			name: "full",
+			obj:  &SSHKeyRef{ID: "a"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func TestNewSSHKeysClient(t *testing.T) {
+	tc := testclient.New(nil, nil, nil)
+	c := NewSSHKeysClient(tc)
+	assert.Equal(t, tc, c.client)
+	assert.Equal(t, &url.URL{Path: "/core/v1/"}, c.basePath)
+}
+
+func TestSSHKeysClient_List(t *testing.T) {
+	type args struct {
+		ctx  context.Context
+		org  OrganizationRef
+		opts *ListOptions
+	}
+	tests := []struct {
+		name     string
+		args     args
+		resp     *katapult.Response
+		respErr  error
+		respV    *sshKeysResponseBody
+		want     []*AuthSSHKey
+		wantReq  *katapult.Request
+		wantResp *katapult.Response
+		wantErr  string
+	}{
+		{
+			name: "success",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+				opts: &ListOptions{
+					Page:    5,
+					PerPage: 32,
+				},
+			},
+			resp: &katapult.Response{
+				Response: &http.Response{StatusCode: http.StatusOK},
+			},
+			respV: &sshKeysResponseBody{
+				Pagination: &katapult.Pagination{Total: 394},
+				SSHKeys: []*AuthSSHKey{
+					{ID: "ssh_O574YEEEYeLmqdmn"},
+				},
+			},
+			want: []*AuthSSHKey{
+				{ID: "ssh_O574YEEEYeLmqdmn"},
+			},
+			wantReq: &katapult.Request{
+				Method: "GET",
+				URL: &url.URL{
+					Path: "/core/v1/organizations/_/ssh_keys",
+					RawQuery: url.Values{
+						"page":     []string{"5"},
+						"per_page": []string{"32"},
+						"organization[id]": []string{
+							"org_O648YDMEYeLmqdmn",
+						},
+					}.Encode(),
+				},
+			},
+			wantResp: &katapult.Response{
+				Pagination: &katapult.Pagination{Total: 394},
+				Response:   &http.Response{StatusCode: http.StatusOK},
+			},
+		},
+		{
+			name: "success with nil options",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			resp: &katapult.Response{
+				Pagination: &katapult.Pagination{Total: 333},
```

Comment:
`Pagination` here should be moved to `respV`, response status should be added, and we should also add `wantResp` field to verify both status and pagination details that come back are correct. It might be slight duplicated effort, but considering it's a test case with a different input to the function, I think it's a good idea ensure results still behave the same way.